src/modules/launcher/Plugins/Community.PowerToys.Run.Plugin.ChatGPT/Main.cs

```diff
@@ -0,0 +1,234 @@
+            }
+            else if (!query.Search.EndsWith("?", StringComparison.OrdinalIgnoreCase))
+            {
+                Result missingQuestionMarkResult = new()
+                {
+                    Title = "End the query with \"?\" to process it",
+                };
+
+                results.Add(missingQuestionMarkResult);
+            }
+            else
+            {
+                Result loadingResult = new()
+                {
+                    Title = "Processing your query...",
+                };
+
+                results.Add(loadingResult);
+            }
+
+            return results;
+        }
+
+        public string GetChatGPTAnswer(string query)
+        {
+            JsonObject data = new JsonObject
+                {
+                    // TODO: Allow to change the model through a setting?
+                    { "model", "gpt-3.5-turbo" },
+                    {
+                        "messages", new JsonArray
+                        {
+                            new JsonObject
+                            {
+                                { "role", "user" },
+                                { "content", query },
+                            },
+                        }
+                    },
+                };
+
+            string url = "https://api.openai.com/v1/chat/completions";
+            string contentType = "application/json";
+            string responseContent = string.Empty;
+
+            using (var httpClient = new HttpClient())
+            {
+                // TODO: Allow to set the custom Auth Key with a setting
+                httpClient.DefaultRequestHeaders.Add("Authorization", PersonalAPIKey);
+
+                var content = new StringContent(data.ToJsonString(), Encoding.UTF8, contentType);
+                var response = httpClient.Send(new HttpRequestMessage(HttpMethod.Post, url)
+                {
+                    Content = content,
+                });
+
+                try
+                {
+                    if (response.IsSuccessStatusCode)
+                    {
+                        JsonDocument doc = JsonDocument.Parse(response.Content.ReadAsStringAsync().Result);
+                        JsonElement root = doc.RootElement;
+                        JsonElement messageElement = root.GetProperty("choices")[0].GetProperty("message");
+                        responseContent = messageElement.GetProperty("content").GetString();
+
+                        // while (responseContent.StartsWith("/n"))
+                        // {
+                        //    responseContent = responseContent.Replace("/n", string.Empty);
+                        // }
+
+                        // TODO: Two whitespaces appear here (gpt returns /n/n at the start), remove them
+                    }
+                    else
+                    {
+                        responseContent = $"Failed: {response.StatusCode} - {response.Content.ReadAsStringAsync().Result}";
```

Comment:
Is there ever a possibility of the content being null?

Reply:
Will have to check. This is v0 code, belongs to the first hacked-up version I made in the first night.